src/components/Editor/TabGate.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+var accessibilityMessage = `Press any key to edit code, press 'Esc' when finished.`;
+var offscreen = {
+  height: '1px',
+  overflow: 'hidden',
+  position: 'absolute',
+  width: '1px',
+};
+
+export default class TabGate extends React.Component {
+  componentDidMount() {
+    hide(this.navTip);
+
+    if (!this.focusTarget) {
+      this.focusTarget =
+        document.getElementById(this.props.idToFocus) ||
+        findFirstPre(this.ref) ||
+        this.ref;
+    }
+
+    //Make all child elements untabbable
+    removeChildTabIndexes(this.ref);
+    
+    var enterHandler = createEnterHandler(this.focusTarget);
+    
+    this.ref.addEventListener('keydown', enterHandler);
+    this.ref.addEventListener('keydown', handleEscape, true);
+    this.ref.addEventListener('focus', event => {
+      addTabGuard(event.target);
+    });
+    this.ref.addEventListener('blur', event => {
+      removeRemoveTabGuard(event.target);
+    });
+  }
+
+  render() {
+    return (
+      <div ref={element => (this.ref = element)} tabIndex="0">
```

Comment:
The problem with this is that it's introducing more elements than what's necessary. I've written down before why this is not the solution I'm searching for. And unfortunately I cannot accept these changes into the library as is, because of these reasons.

Sorry :/

On the other hand, why give up with the previous PR when it was close to being correct? I can probably take over that PR at some point if you're unclear on what went wrong, but maybe it'd be better to get a bit of distance from this first, and to think of why it was buggy in the first place.